### ISSUE TITLE
Site editor has no visual or screenreader notifications about the result of saving

### ISSUE THREAD

**Issue by talldan** [WordPress/gutenberg#36166]
### Description

When saving in the site editor, there's no visual notification (snackbar message) or screenreader message to indicate saving has completed successfully or unsuccessfully (like when offline). I think this needs to be solved before shipping the editor.

A snackbar (on success) or error notice (on failure) should be quick wins that would solve this.

A second part that could be improved is the entity saving panel. When clicking the first 'Save' button, a second save button is focused and testing through Voiceover I just head 'Button, Save, publish' again, which sounds buggy, almost like the first button didn't do anything. It could be an improvement to add a description like 'Confirm your changes' to this button. At least something that indicates there is more content below the second save button.

### Step-by-step reproduction instructions

With a screenreader active:
1. Open the site editor
2. Make some changes to the content
3. Focus the save button and hit enter.
4. Hear that a save button has been focused again, but what actually happened after clicking the first button is a bit hard to follow
5. Optionally set network throttling offline in the browser dev tools
6. Press save again
7. Observe there's no message indicating whether saving was successful or unsuccessful

### Environment info

Tested in Voiceover / Safari
Tested network throttling in Brave, since Safari doesn't have this feature 😞 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
